### PR TITLE
Waiting queue

### PR DESCRIPTION
**Pull request by agnessaik**
I am thinking about better waiting queue column-row logic. Is WaitingQueueScene class structure logical? 
So I am waiting for suggestions and would like to test if that works with correct data types and server.

### CODE REVIEW

File: client/src/main/java/Render.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    private static Stage primaryStage;
+
+    public static Render getInstance() {
+        return ourInstance;
+    }
+
+    @Override
+    public void start(Stage primaryStage) throws IOException {
+        this.primaryStage = primaryStage;
+        // TODO: open user login scene
```
Comment (KeijoTomiste) [inline client/src/main/java/Render.java]: think login should be initiated where this class is initiated, after this is initiated

File: client/src/main/java/services/ServerCommunicationService.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package services;
+
+import io.reactivex.Observable;
```
Comment (KeijoTomiste) [inline client/src/main/java/services/ServerCommunicationService.java]: does your code even run? i dont see `reactivex` being added

File: client/src/main/java/WaitingQueueScene.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    private void createWaitingQueueSceneAndKeepItUpdated(){
+        setWaitingQueueSceneBase();
+        WaitingQueueService.getInstance().getWaitingQueue().subscribe(data -> {
+            //addAllWaiters(gridPane, data);
+        });
+    }
+
+    private void addAllWaiters(GridPane gridPane, List<String> usersWaiting) {
```
Comment (KeijoTomiste) [inline client/src/main/java/WaitingQueueScene.java]: isn't `gridPane` excessive? arent you supposed to use the one on the class?

File: client/src/main/java/WaitingQueueScene.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        if (gridPane.getChildren() != null){
```
Comment (KeijoTomiste) [inline client/src/main/java/WaitingQueueScene.java]: you sure that this can even return null?

File: client/src/main/java/WaitingQueueScene.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        createWaitingQueueSceneAndKeepItUpdated();
+    }
+
+    public void setWaitingQueueSceneBase() {
+        GridPane gridPane = createGridPane();
+        Scene waitingQueueScene = new Scene(gridPane);
+        Render.getInstance().showScene(waitingQueueScene);
+    }
+
+    private void createWaitingQueueSceneAndKeepItUpdated(){
```
Comment (KeijoTomiste) [inline client/src/main/java/WaitingQueueScene.java]: maaan this is a long name :) maybe `createAndUpdateScene`

File: client/src/main/java/Render.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        // first scene user login screen, if user has registered;
+        // that scene opens WaitingQueueScene;
+        // if game starts renders battlefield
+        new WaitingQueueScene();
```
Comment (KeijoTomiste) [inline client/src/main/java/Render.java]: dont think that should be opened here anyways, login initiates your component or sth like that

File: client/src/main/java/WaitingQueueScene.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        setWaitingQueueSceneBase();
+        WaitingQueueService.getInstance().getWaitingQueue().subscribe(data -> {
+            //addAllWaiters(gridPane, data);
```
Comment (KeijoTomiste) [inline client/src/main/java/WaitingQueueScene.java]: could have left this in

**Comment by TaliKert** [comment]
I noticed that in WaitingQueueScene, createAndUpdateScene() calls out setWaitingQueueSceneBase(), which creates a new gridpane each time and also resets the renderer scene with showScene. Both of these could be done a single time and from there on making changes only to the gridpane.

**Pull request merged**